test/weights.jl

```diff
@@ -3,355 +3,376 @@ using Base.Test
-@test r ≈ [dot(x, w)]
+        r = ones(1)
+        @test wsum!(r, x, w, 1; init=true) === r
+        @test r ≈ [dot(x, w)]
 
-r = ones(1)
-@test wsum!(r, x, w, 1; init=false) === r
-@test r ≈ [dot(x, w) + 1.0]
+        r = ones(1)
+        @test wsum!(r, x, w, 1; init=false) === r
+        @test r ≈ [dot(x, w) + 1.0]
 
-x = rand(6, 8)
-w1 = rand(6)
-w2 = rand(8)
+        x = rand(6, 8)
+        w1 = rand(6)
+        w2 = rand(8)
 
-r = ones(1, 8)
-@test wsum!(r, x, w1, 1; init=true) === r
-@test r ≈ sum(x .* w1, 1)
+        r = ones(1, 8)
+        @test wsum!(r, x, w1, 1; init=true) === r
+        @test r ≈ sum(x .* w1, 1)
 
-r = ones(1, 8)
-@test wsum!(r, x, w1, 1; init=false) === r
-@test r ≈ sum(x .* w1, 1) .+ 1.0
+        r = ones(1, 8)
+        @test wsum!(r, x, w1, 1; init=false) === r
+        @test r ≈ sum(x .* w1, 1) .+ 1.0
 
-r = ones(6)
-@test wsum!(r, x, w2, 2; init=true) === r
-@test r ≈ sum(x .* w2', 2)
+        r = ones(6)
+        @test wsum!(r, x, w2, 2; init=true) === r
+        @test r ≈ sum(x .* w2', 2)
 
-r = ones(6)
-@test wsum!(r, x, w2, 2; init=false) === r
-@test r ≈ sum(x .* w2', 2) .+ 1.0
+        r = ones(6)
+        @test wsum!(r, x, w2, 2; init=false) === r
+        @test r ≈ sum(x .* w2', 2) .+ 1.0
 
-x = rand(8, 6, 5)
-w1 = rand(8)
-w2 = rand(6)
-w3 = rand(5)
+        x = rand(8, 6, 5)
+        w1 = rand(8)
+        w2 = rand(6)
+        w3 = rand(5)
 
-r = ones(1, 6, 5)
-@test wsum!(r, x, w1, 1; init=true) === r
-@test r ≈ sum(x .* w1, 1)
+        r = ones(1, 6, 5)
+        @test wsum!(r, x, w1, 1; init=true) === r
+        @test r ≈ sum(x .* w1, 1)
 
-r = ones(1, 6, 5)
-@test wsum!(r, x, w1, 1; init=false) === r
-@test r ≈ sum(x .* w1, 1) .+ 1.0
+        r = ones(1, 6, 5)
+        @test wsum!(r, x, w1, 1; init=false) === r
+        @test r ≈ sum(x .* w1, 1) .+ 1.0
 
-r = ones(8, 1, 5)
-@test wsum!(r, x, w2, 2; init=true) === r
-@test r ≈ sum(x .* w2', 2)
+        r = ones(8, 1, 5)
+        @test wsum!(r, x, w2, 2; init=true) === r
+        @test r ≈ sum(x .* w2', 2)
 
-r = ones(8, 1, 5)
-@test wsum!(r, x, w2, 2; init=false) === r
-@test r ≈ sum(x .* w2', 2) .+ 1.0
+        r = ones(8, 1, 5)
+        @test wsum!(r, x, w2, 2; init=false) === r
+        @test r ≈ sum(x .* w2', 2) .+ 1.0
 
-r = ones(8, 6)
-@test wsum!(r, x, w3, 3; init=true) === r
-@test r ≈ sum(x .* reshape(w3, (1, 1, 5)), 3)
+        r = ones(8, 6)
+        @test wsum!(r, x, w3, 3; init=true) === r
+        @test r ≈ sum(x .* reshape(w3, (1, 1, 5)), 3)
 
-r = ones(8, 6)
-@test wsum!(r, x, w3, 3; init=false) === r
-@test r ≈ sum(x .* reshape(w3, (1, 1, 5)), 3) .+ 1.0
+        r = ones(8, 6)
+        @test wsum!(r, x, w3, 3; init=false) === r
+        @test r ≈ sum(x .* reshape(w3, (1, 1, 5)), 3) .+ 1.0
+    end
+end
 
+@testset "Sum and mean syntax" begin
+    a = reshape(1.0:27.0, 3, 3, 3)
 
-## the sum and mean syntax
+    @testset "Sum" begin
+        @test sum([1.0, 2.0, 3.0], fweights([1.0, 0.5, 0.5])) ≈ 3.5
```

Comment:
Unless some tests are really redundant, I'd rather have too many tests than too few of them.